src/minimum_tracer/minimum_tracer.cpp

```diff
@@ -0,0 +1,3214 @@
+  {
+    for (auto pair : flat_2D_dirs)
+    {
+      auto a               = point.at(pair.at(0));
+      auto b               = point.at(pair.at(1));
+      auto c               = std::sqrt(a * a + b * b);
+      point.at(pair.at(0)) = c;
+      point.at(pair.at(1)) = 0;
+    }
+  }
+
+  return;
+}
+
+void MinimumTracer::FindDiscreteSymmetries()
+{
+  const double GroupElementslMaximumRelativeError =
+      1e-8; // Maximum value of |V/GroupElements(V)-1|
+  const size_t NumberOfGroupElementsChecks =
+      10; // Number of times GroupElements is checked
+  const size_t dim = modelPointer->get_nVEV(); // Number of VEVs
+
+  std::function<double(Eigen::VectorXd)> V = [&](Eigen::VectorXd vev)
+  {
+    // Potential wrapper at T=0 for tree-level potential
+    std::vector<double> res = this->modelPointer->MinimizeOrderVEV(
+        std::vector<double>(vev.data(), vev.data() + vev.size()));
+    return this->modelPointer->VEff(res, 0, 0, 0);
+  };
+
+  // Generate random VEV
+  srand(time(NULL));
+  const long max_rand                                = 1000000L;
+  std::function<Eigen::VectorXd()> GenerateRandomVEV = [&]()
+  {
+    Eigen::VectorXd res(dim);
+    for (auto &m : res)
+      m = 1. + 299. * (rand() % max_rand) / max_rand;
+    return res;
+  };
+
+  // Check if potential is invariant under group element
+  std::function<int(Eigen::MatrixXd)> CheckGroupElement =
+      [&](Eigen::MatrixXd GroupElement)
+  {
+    int result = 1;
+    for (std::size_t it = 0; it < NumberOfGroupElementsChecks; it++)
+    {
+      auto RandomVEV = GenerateRandomVEV();
+      result *= (abs(V(RandomVEV) / V(GroupElement * RandomVEV) - 1) <
+                 GroupElementslMaximumRelativeError);
+    }
+    return result;
+  };
+
+  // Store available possible
+  std::vector<Eigen::MatrixXd> C2GroupElements, SnGroupElements,
+      StoreGroupElements;
+
+  // n-plet of C2
+  for (std::size_t C2Nplet = 0; C2Nplet <= dim; C2Nplet++)
+  {
+    // Construct bitmask. bx11...000
+    std::string bitmask(C2Nplet, 1); // C2Nplet leading 1's
+    bitmask.resize(dim, 0);          // NVEV - C2Nplet trailing 0's
+    do                               // Permute bitmask
+    {
+      // Start with identity matrix
+      Eigen::MatrixXd C2_element = MatrixXd::Identity(dim, dim);
+      for (std::size_t i = 0; i < dim; ++i) // Flip i element
+      {
+        if (bitmask[i]) C2_element(i, i) = -1;
+      }
+      C2GroupElements.push_back(C2_element);
+    } while (std::prev_permutation(bitmask.begin(), bitmask.end()));
+  }
+
+  /*
+    Currently permutations of VEV is not working
+
+    Eigen::PermutationMatrix<Dynamic, Dynamic> PermutationMatrix(dim);
+    PermutationMatrix.setIdentity();
+    do
+    {
+      SnGroupElements.push_back(Eigen::MatrixXd(PermutationMatrix));
+    } while (std::next_permutation(PermutationMatrix.indices().begin(),
+                                   PermutationMatrix.indices().end()));
+                                   */
```

Comment:
Is this a problem? Maybe make a issue out of it

Reply:
We haven't finished developing the math yet. Permutations make the group non-abelian which breaks things. I will open an issue so that this gets implemented in the future.